### ISSUE TITLE
Filesystem is not idempotent with xfs

### ISSUE THREAD

**Issue by tedwardd** [ansible-collections/community.general#5739]
### Summary

Using `community.general.filesyste` with `fstype: xfs` is not idempotent

### Issue Type

Bug Report

### Component Name

filesystem

### Ansible Version

```console (paste below)
$ ansible --version
ansible [core 2.13.6]
  config file = /Users/tedwardd/git-work/personal/ansible-playbooks/ansible.cfg
  configured module search path = ['/Users/tedwardd/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /opt/homebrew/Cellar/ansible/6.6.0/libexec/lib/python3.11/site-packages/ansible
  ansible collection location = /Users/tedwardd/.ansible/collections:/usr/share/ansible/collections
  executable location = /opt/homebrew/bin/ansible
  python version = 3.11.0 (main, Oct 25 2022, 13:57:33) [Clang 14.0.0 (clang-1400.0.29.202)]
  jinja version = 3.1.2
  libyaml = True
```

### Community.general Version

```console (paste below)
$ ansible-galaxy collection list community.general

# /Users/tedwardd/.ansible/collections/ansible_collections
Collection        Version
----------------- -------
community.general 6.1.0  

# /opt/homebrew/Cellar/ansible/6.6.0/libexec/lib/python3.11/site-packages/ansible_collections
Collection        Version
----------------- -------
community.general 5.8.0  
```

### Configuration

```console (paste below)
$ ansible-config dump --only-changed
DEFAULT_HOST_LIST(/Users/tedwardd/git-work/personal/ansible-playbooks/ansible.cfg) = ['/Users/tedwardd/git-work/personal/ansible-playbooks/hosts.yaml']
DEFAULT_LOG_PATH(/Users/tedwardd/git-work/personal/ansible-playbooks/ansible.cfg) = /Users/tedwardd/git-work/personal/ansible-playbooks/out.log
```

### OS / Environment

Ubuntu 22.04

### Steps to Reproduce

```yaml (paste below)
- name: Create plotter temp storage
      block:
        - name: Create new partition
          community.general.parted:
            device: /dev/sdj
            number: 1
            state: present
            fs_type: xfs
        - name: Create fs
          community.general.filesystem:
            fstype: xfs
            dev: /dev/sdj1
            state: present
        - name: Create mount path
          ansible.builtin.file:
            path: /mnt/example
            state: directory
            owner: my_user
            group: my_user
            mode: '0755'
        - name: Mount partition
          ansible.posix.mount:
            path: /mnt/example
            src: /dev/sdj1
            fstype: xfs
            state: mounted
```

### Expected Results

Task, "Create fs" should detect the existence of an xfs partition on /dev/sdj1 and pass

### Actual Results

```console (paste below)
TASK [Create fs] ******************************************************************************************************************************
fatal: [jupiter]: FAILED! => {"changed": false, "cmd": "/usr/sbin/mkfs.xfs -f /dev/sdj1", "msg": "mkfs.xfs: /dev/sdj1 contains a mounted filesystem\nUsage: mkfs.xfs\n/* blocksize */\t\t[-b size=num]\n/* config file */\t[-c options=xxx]\n/* metadata */\t\t[-m crc=0|1,finobt=0|1,uuid=xxx,rmapbt=0|1,reflink=0|1,\n\t\t\t    inobtcount=0|1,bigtime=0|1]\n/* data subvol */\t[-d agcount=n,agsize=n,file,name=xxx,size=num,\n\t\t\t    (sunit=value,swidth=value|su=num,sw=num|noalign),\n\t\t\t    sectsize=num\n/* force overwrite */\t[-f]\n/* inode size */\t[-i perblock=n|size=num,maxpct=n,attr=0|1|2,\n\t\t\t    projid32bit=0|1,sparse=0|1]\n/* no discard */\t[-K]\n/* log subvol */\t[-l agnum=n,internal,size=num,logdev=xxx,version=n\n\t\t\t    sunit=value|su=num,sectsize=num,lazy-count=0|1]\n/* label */\t\t[-L label (maximum 12 characters)]\n/* naming */\t\t[-n size=num,version=2|ci,ftype=0|1]\n/* no-op info only */\t[-N]\n/* prototype file */\t[-p fname]\n/* quiet */\t\t[-q]\n/* realtime subvol */\t[-r extsize=num,size=num,rtdev=xxx]\n/* sectorsize */\t[-s size=num]\n/* version */\t\t[-V]\n\t\t\tdevicename\n<devicename> is required unless -d name=xxx is given.\n<num> is xxx (bytes), xxxs (sectors), xxxb (fs blocks), xxxk (xxx KiB),\n      xxxm (xxx MiB), xxxg (xxx GiB), xxxt (xxx TiB) or xxxp (xxx PiB).\n<value> is xxx (512 byte blocks).", "rc": 1, "stderr": "mkfs.xfs: /dev/sdj1 contains a mounted filesystem\nUsage: mkfs.xfs\n/* blocksize */\t\t[-b size=num]\n/* config file */\t[-c options=xxx]\n/* metadata */\t\t[-m crc=0|1,finobt=0|1,uuid=xxx,rmapbt=0|1,reflink=0|1,\n\t\t\t    inobtcount=0|1,bigtime=0|1]\n/* data subvol */\t[-d agcount=n,agsize=n,file,name=xxx,size=num,\n\t\t\t    (sunit=value,swidth=value|su=num,sw=num|noalign),\n\t\t\t    sectsize=num\n/* force overwrite */\t[-f]\n/* inode size */\t[-i perblock=n|size=num,maxpct=n,attr=0|1|2,\n\t\t\t    projid32bit=0|1,sparse=0|1]\n/* no discard */\t[-K]\n/* log subvol */\t[-l agnum=n,internal,size=num,logdev=xxx,version=n\n\t\t\t    sunit=value|su=num,sectsize=num,lazy-count=0|1]\n/* label */\t\t[-L label (maximum 12 characters)]\n/* naming */\t\t[-n size=num,version=2|ci,ftype=0|1]\n/* no-op info only */\t[-N]\n/* prototype file */\t[-p fname]\n/* quiet */\t\t[-q]\n/* realtime subvol */\t[-r extsize=num,size=num,rtdev=xxx]\n/* sectorsize */\t[-s size=num]\n/* version */\t\t[-V]\n\t\t\tdevicename\n<devicename> is required unless -d name=xxx is given.\n<num> is xxx (bytes), xxxs (sectors), xxxb (fs blocks), xxxk (xxx KiB),\n      xxxm (xxx MiB), xxxg (xxx GiB), xxxt (xxx TiB) or xxxp (xxx PiB).\n<value> is xxx (512 byte blocks).\n", "stderr_lines": ["mkfs.xfs: /dev/sdj1 contains a mounted filesystem", "Usage: mkfs.xfs", "/* blocksize */\t\t[-b size=num]", "/* config file */\t[-c options=xxx]", "/* metadata */\t\t[-m crc=0|1,finobt=0|1,uuid=xxx,rmapbt=0|1,reflink=0|1,", "\t\t\t    inobtcount=0|1,bigtime=0|1]", "/* data subvol */\t[-d agcount=n,agsize=n,file,name=xxx,size=num,", "\t\t\t    (sunit=value,swidth=value|su=num,sw=num|noalign),", "\t\t\t    sectsize=num", "/* force overwrite */\t[-f]", "/* inode size */\t[-i perblock=n|size=num,maxpct=n,attr=0|1|2,", "\t\t\t    projid32bit=0|1,sparse=0|1]", "/* no discard */\t[-K]", "/* log subvol */\t[-l agnum=n,internal,size=num,logdev=xxx,version=n", "\t\t\t    sunit=value|su=num,sectsize=num,lazy-count=0|1]", "/* label */\t\t[-L label (maximum 12 characters)]", "/* naming */\t\t[-n size=num,version=2|ci,ftype=0|1]", "/* no-op info only */\t[-N]", "/* prototype file */\t[-p fname]", "/* quiet */\t\t[-q]", "/* realtime subvol */\t[-r extsize=num,size=num,rtdev=xxx]", "/* sectorsize */\t[-s size=num]", "/* version */\t\t[-V]", "\t\t\tdevicename", "<devicename> is required unless -d name=xxx is given.", "<num> is xxx (bytes), xxxs (sectors), xxxb (fs blocks), xxxk (xxx KiB),", "      xxxm (xxx MiB), xxxg (xxx GiB), xxxt (xxx TiB) or xxxp (xxx PiB).", "<value> is xxx (512 byte blocks)."], "stdout": "", "stdout_lines": []}
```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

**Comment by ansibullbot** [comment]
Files identified in the description:
* [`plugins/modules/filesystem.py`](https://github.com/ansible-collections/community.general/blob/main/plugins/modules/filesystem.py)

If these files are incorrect, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/devel/ISSUE_HELP.md)

**Comment by ansibullbot** [comment]
cc @abulimov @pilou- @quidame
[click here for bot help](https://github.com/ansible/ansibullbot/blob/devel/ISSUE_HELP.md)

**Comment by tedwardd** [comment]
In doing some additional testing, it appears that changing:
```yaml
        - name: Create fs
          community.general.filesystem:
            fstype: xfs
            dev: /dev/sdj1
            state: present
```
to
```yaml
        - name: Create fs
          community.general.filesystem:
            fstype: xfs
            dev: /dev/sdj1
            state: present
            resizefs: true
```

behaves in an idempotent manner but leaving the default or setting `resizefs: false` results in `community.general.filesystem` re-creating the filesystem on each run regardless of whether or not it exists. If the target filesystem is mounted, however, this produces an error since you can not target a mounted device when creating a filesystem.

**Comment by tedwardd** [comment]
OK, I'm closing this because in my continued testing of this issue I ended up figuring out what I believe to be the issue.

If you look at my original report, the output of `ansible-galaxy collection list community.general` shows two versions installed. On another system I use, however, I only have `6.1.0` installed and I noticed that none of the issues I was experiencing occur on this system. I believe the issue is that, despite having `6.1.0` installed, my system is preferring to use the older `5.8.0` release which, after reviewing the code, does not contain any of the bits added to detect the existence of an XFS filesystem prior to attempting to create the one defined in the playbook.
